File: INF-143/contest_8/E/main.cpp
#include<bits/stdc++.h>

using namespace std;

int main()
{
	ios_base::sync_with_stdio(false);
	cin.tie(NULL);
	freopen("input.txt", "r", stdin);
	string cipher, line, salida;

	vector<string> salidas;

	while(cin >> cipher)
	{
		map<char, char> mapa;

		line = salida = "";		

		for(int i = 0, chr = 97; i < cipher.size(); i++, chr++)
			mapa[cipher[i]] = chr;
		mapa[' '] = ' ';
		
		int n;

		cin >> n;

		n++;

		while(n--)
		{
			getline(cin, line, '\n');

			for(int i = 0; i < line.size(); i++)
			{
				salida += mapa[line[i]];
			}

			salida += "\n";
		}

		salidas.push_back(salida.substr(1, salida.size()-1));
	}

	for(int i = 0; i < salidas.size() - 1; i++)
		cout << salidas[i] << "\n";
	cout << salidas[salidas.size() - 1];

	return 0;
}

File: INF-143/contest_8/E/main.py
salidas = list()

def sol():
	cipher = input()

	mapa = dict()

	x = 97

	for i in cipher:
		mapa[i] = chr(x)
		x += 1
	mapa[' '] = ' '
	n = int(input())
	salida = str()

	for _ in range(n):
		line = input()
		for i in line:
			salida += mapa[i]
		salida += "\n"

	salidas.append(salida)
while True:
	try:
		sol()
	except EOFError:
		break
"""
salida = str()
for i in range(len(salidas) - 1):
	salida += salidas[i]
salida += salidas[len(salidas) - 1][0:len(salidas[len(salidas) - 1]) - 1]
print(salida)
"""